### PR TITLE
CC-12: created reusable component and added propTypes

### PR DESCRIPTION
**Pull request by StephanieKirshner**
frontEnd
backend

### CODE REVIEW

File: src/components/HomePage/HomePage.jsx [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
 /* eslint-disable no-unused-vars */
 import React, { useState } from 'react';
-
+import PropTypes from 'prop-types';
 import './style.css';
 
-const HomePage = ({ data }) => {
-  const [modalType, setModalType] = useState(0);
+const HomePage = ({ data, buttonsToRender }) => {
+  const [modalType, setModalType, createNew] = useState(0);
```
Comment (jdickerson-banyan) [inline src/components/HomePage/HomePage.jsx:8]: Remove createNew

File: src/components/HomePage/HomePage.jsx [hunk 2]
```diff
@@ -1,27 +1,36 @@
...
+          <button
+            className='signInBtn'
+            onClick={() => setModalType(2)}
+            style={{ backgroundColor: '#9fb5d5' }}
+          >
+            Sign In
+          </button>
+        </div>
+      ) : (
+        <button className='createNew' onClick={createNew}>
```
Comment (jdickerson-banyan) [inline src/components/HomePage/HomePage.jsx:29]: remove the onClick please

File: src/components/HomePage/HomePage.jsx [hunk 3]
```diff
@@ -46,4 +55,16 @@ const HomePage = ({ data }) => {
   );
 };
 
+HomePage.propTypes = {
+  data: PropTypes.array,
+};
+
+HomePage.propTypes = {
+  buttonsToRender: PropTypes.string,
+};
```
Comment (jdickerson-banyan) [inline src/components/HomePage/HomePage.jsx:64]: combine the object for data and buttonsToRender and remove the propTypes for modalType (it is a state value, not props)

**Review by jdickerson-banyan — CHANGES_REQUESTED** [review]

File: src/components/HomePage/HomePage.jsx (hunk 1 above)
Comment (StephanieKirshner) [reply src/components/HomePage/HomePage.jsx:8]: done

File: src/components/HomePage/HomePage.jsx (hunk 2 above)
Comment (StephanieKirshner) [reply src/components/HomePage/HomePage.jsx:29]: done

File: src/components/HomePage/HomePage.jsx (hunk 3 above)
Comment (StephanieKirshner) [reply src/components/HomePage/HomePage.jsx:64]: done

**Review by jdickerson-banyan — APPROVED** [review]

**Pull request merged**